features/environment.py

```diff
@@ -141,33 +162,31 @@ def _capture_container_as_image(container_name: str, image_name: str) -> None:
     subprocess.run(["lxc", "publish", container_name, "--alias", image_name])
 
 
-def create_trusty_uat_lxd_image(context: Context) -> None:
-    """Create a trusty lxd image with ubuntu-advantage-tools installed
+def create_uat_lxd_image(context: Context, series: str) -> None:
+    """Create a given series lxd image with ubuntu-advantage-tools installed
 
     This will launch a container, install ubuntu-advantage-tools, and publish
     the image.  The image's name is stored in context.image_name for use within
     step code.
 
     :param context:
         A `behave.runner.Context`; this will have `image_name` set on it.
+    :param series:
+       A string representing the series name to create
     """
 
-    def image_cleanup() -> None:
-        if context.config.image_clean:
-            subprocess.run(["lxc", "image", "delete", context.image_name])
-        else:
-            print("Image cleanup disabled, not deleting:", context.image_name)
-
     if context.reuse_container:
         print(" Reusing the existent container: ", context.reuse_container)
     else:
         now = datetime.datetime.now()
-        context.image_name = "behave-image-" + now.strftime("%s%f")
-        build_container_name = "behave-image-build-" + now.strftime("%s%f")
-        launch_lxd_container(context, "ubuntu:trusty", build_container_name)
+        context.image_name = "behave-image-%s-" % series + now.strftime("%s%f")
```

Comment:
I think you need to avoid setting image_name here and just set context.series_image_names[series]
```suggestion
        context.series_image_names[series] = "behave-image-%s-" % series + now.strftime("%s%f")
```

Reply:
Thanks. I changed it.